### ISSUE TITLE
Dead-code-elimination: Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by Naddiseo** [babel/babel#931]
As a follow up to fixing my previous issue #930, I noticed the DCE transformer dies with the following:

```
Module build failed: TypeError: ~/projects/test/node_modules/babel-runtime/core-js.js: Cannot read property 'length' of undefined
    at TraversalPath.exit (~/projects/test/node_modules/babel-core/lib/babel/transformation/transformers/utility/dead-code-elimination.js:86:57)
    at TraversalPath.call (~/projects/test/node_modules/babel-core/lib/babel/traversal/path.js:80:26)
    at TraversalPath.visit (~/projects/test/node_modules/babel-core/lib/babel/traversal/path.js:117:12)
    at TraversalConext.visitNode (~/projects/test/node_modules/babel-core/lib/babel/traversal/context.js:31:22)
    at TraversalConext.visit (~/projects/test/node_modules/babel-core/lib/babel/traversal/context.js:48:28)
    at Function.traverse.node (~/projects/test/node_modules/babel-core/lib/babel/traversal/index.js:42:17)
    at TraversalPath.visit (~/projects/test/node_modules/babel-core/lib/babel/traversal/path.js:116:16)
    at TraversalConext.visitNode (~/projects/test/node_modules/babel-core/lib/babel/traversal/context.js:31:22)
    at TraversalConext.visit (~/projects/test/node_modules/babel-core/lib/babel/traversal/context.js:39:19)
    at Function.traverse.node (~/projects/test/node_modules/babel-core/lib/babel/traversal/index.js:42:17)
    at TraversalPath.visit (~/projects/test/node_modules/babel-core/lib/babel/traversal/path.js:116:16)
    at TraversalConext.visitNode (~/projects/test/node_modules/babel-core/lib/babel/traversal/context.js:31:22)
    at TraversalConext.visit (~/projects/test/node_modules/babel-core/lib/babel/traversal/context.js:48:28)
```

I've traced it down to using the 'runtime' transformer and the `class`; it seems that if webpack cannot find `babel-runtime` the DCE transformer dies on its output:

```javascript
var _babelHelpers = __webpack_require__(!(function webpackMissingModule() {
    var e = new Error("Cannot find module \"babel-runtime/helpers\"");
    e.code = 'MODULE_NOT_FOUND';
    throw e; 
}()))["default"];

```

**Comment by sebmck** [comment]
I couldn't manage to replicate this. However I added some checks for the `body` existence before `body.length` is checked which should fix this.

**Comment by sebmck** [comment]
Although I'm still worried about what caused this in the first place.

**Comment by neVERberleRfellerER** [comment]
@sebmck I have managed to replicate it. I have some work now, but I'll try to find out where is it coming from (the cause, not the line in stacktrace) or at last cut everything down and create shortest code that triggers it.

**Comment by sebmck** [comment]
@neVERberleRfellerER Rad, thanks!

**Comment by neVERberleRfellerER** [comment]
@sebmck I have it! Ready?
First, simplest code that trigger the error:
```
if (x) 1;
```

And my guess where is it coming from?
https://github.com/babel/babel/blob/master/src/babel/transformation/transformers/utility/dead-code-elimination.js#L80

```
if (t.isBlockStatement(consequent) ...
```
"is" is missing

**Comment by sebmck** [comment]
@neVERberleRfellerER Oh shit, yeah, that'd do it. Thanks!

**Comment by neVERberleRfellerER** [comment]
@sebmck Glad to help! And that error made my day (well, night).

> Oh shit

Same reaction.

**Comment by sebmck** [comment]
Sorry for the delay. Fixed as of 4.6.6, thanks!
